server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -796,11 +796,25 @@ public void batchDelete(Collection<Entitlement> entitlements) {
      * preferably by calling {@link #unlinkEntitlements} first.
      *
      * @param entitlementIds The collection of ids of the entitlements to be deleted.
+     *
+     * @return
+     *  the number of entitlements deleted as a result of this operation
      */
-    public void batchDeleteByIds(Collection<String> entitlementIds) {
-        Map<String, Object> criteria = new HashMap<>();
-        criteria.put("id", entitlementIds);
-        this.bulkSQLDelete(Entitlement.DB_TABLE, criteria);
+    public int batchDeleteByIds(Collection<String> entitlementIds) {
```

Comment:
Could you add a unit-test for this change? I think it would be helpful to know that we do not re-introduce the issue in case we need to make changes in the future.

Reply:
Done